### DISCUSSION TITLE
Installing Kineticstoolkit

### DISCUSSION THREAD

**Discussion by M3nthos** [kineticstoolkit/kineticstoolkit#164] [1. Users Q&A] [answered]
Hello, 

me and a lot of colleges at my university cant install the Kit , 
we have a anaconda-navigator with Python 5.3.3 and couldn't find the problem. 
is there any tip u have or is this a common problem  ? 

greetings and thanks in advance for any help

**Comment by felixchenier** [comment, marked as answer]
Hello @M3nthos, yes we'll find a way to install it. There is no version 5.3.3 of Python yet, but I assume you mean Spyder 5.3.3.

If you installed the whole Anaconda suite and you normally run Spyder from it, then the easiest way to install Kinetics Toolkit is probably directly into the base anaconda environment:

Are you on Windows ? If you are, from the Anaconda menu, run "Anaconda Prompt". In the prompt, write this information as in the screenshot. Press "y" if it asks you to confirm.

<img width="558" alt="image" src="https://user-images.githubusercontent.com/34967663/233193569-71cd59ba-25e1-429a-98c5-167785ee6943.png">

---

While you try this, it would help to know more about the error you get. Is it an installation error, or you are not able to import it once it's installed?

**Reply by M3nthos** [reply]
hello , Thanks to the quick answer , i am working with MacOS

**Reply by felixchenier** [reply]
It would help to know more about the error you get. Is it an installation error, or you are not able to import it once it's installed? What are the steps that you take, exactly, to install the package? Could you also put a screenshot of the error?

**Reply by M3nthos** [reply]
hello , I luckily could solve it by my own right after posting it (sorry for taking ur time :D )
it always stuck at : 
Collecting package metadata (current_repodata.json): 

I just installed a new Spyder Version with 5.4.1 , that it didn't show in the Anaconda Navigator, and its now working with no problem......

i will tell this to all my student colleges here in Germany, we are grateful for ur fast answer and happy that we can now start working on our Projekt.

**Reply by felixchenier** [reply]
Perfect! Don't hesitate to ask for any help.

**Comment by MuhammadUtoro** [comment]
Hello everyone!
I was about to install the kineticstoolkit package and I installed using pip install. Is it the correct way to do it? I am also using visual studio code. However, as I wanted to run the example, the modules cannot be found. Can I use visual studio code? or Do I have to use spyder?
Thank you for your help.

Regards,
Muhammad

**Reply by felixchenier** [reply]
Hi @MuhammadUtoro 

> I was about to install the kineticstoolkit package and I installed using pip install. Is it the correct way to do it? 

It is a correct way to do it. However, you won't be able to read or write c3d files, because we rely on ezc3d which is distributed only conda-forge. If you need to read/write c3d files, then you must install ezc3d, using conda. But in this case, it may be a better idea to also install kinetics toolkit using conda, which automatically installs ezc3d.

> Can I use visual studio code?

Yes you can. But you need to select the correct Python interpreter that corresponds to where you installed kineticstoolkit in a first place. To do this, run the command palette (on a mac: cmd+shift+p, I guess it's ctrl+shift+p on Windows) and select "Python: Select interpreter" then select the right Python interpreter. I couldn't help you more on Visual Studio Code because I very rarely use it with Python.

> Do I have to use spyder?

Personally I prefer Spyder for science-related work using Python, but you absolutely don't need it.

**Comment by MuhammadUtoro** [comment]
Hi @felixchenier 
I see. Now, that makes sense why I can't read the c3d files and showing the error. You absolutely right and I have installed kinetics tool kit using conda and it is working perfectly fine. I might switch to spyder as well as you suggested.
Thanks a lot for your assistance and suggestion.

Regards,
Muhammad
